### PR TITLE
Static/dynamic library products

### LINKED ISSUE #159: How to link this as the static framework(or library) with SPM?
First, this is really nice & useful for me to develop with SwiftUI through SPM! Thank you for creating!!!!

Well, recently I've been trying to use the Xcode cloud(it's on the Beta status, though). 
Then, there is one trouble to link the Introspect. It's below.

> ITMS-90334: Invalid Code Signature Identifier. The identifier "Introspect-55554944a57ff822a61e381886dc5802237a083e" in your code signature for "Introspect_359AFEED79E48935_PackageProduct" must match its Bundle Identifier "Introspect"

Since I found out, the issue seems to be from linking the dynamic framework. It occurs,  even though, the Introspect doesn't have the `type` property as `dynamic`.
(This is talking about the Package.swift's setting properties)

So, I'd like to know how to use the Introspect as the static framework through SPM.
Because they're linked as the static framework, even though some other frameworks(ex. Alamofire) doesn't have any settings for this, through SPM. Is there any specific options using frameworks or libraries through SPM?

I've had one thing to resolve this. It's the Introspect target's `MACH_O_TYPE` property sets the `static library` in the Introspect project. 
It's useful & effective to avoid the trouble. So, I'm preparing the P.R for it. But, I think there could be the other awesome resolution.
So, I'd like to ask a question like this title.
Could anyone help or give an advice for this?

### PR DESCRIPTION
**Pull request by krzyzanowskim**
Resolves #159 

This workaround Xcode Cloud builds (due to longstanding Xcode Cloud bug that can't build and link dynamic dependency properly)

### CODE REVIEW

**Review by jegumhon — APPROVED** [review]

**Comment by krzyzanowskim** [comment]
who's in power to merge it?

File: Package.swift [hunk 1]
```diff
@@ -12,6 +12,7 @@ let package = Package(
     products: [
         .library(
             name: "Introspect",
+            type: .static,
             targets: ["Introspect"]
         )
```
Comment (davdroman) [inline Package.swift:17]: I don't think we can accept such a big breaking change. Though the following might be acceptable:

```suggestion
        .library(name: "Introspect", targets: ["Introspect"]),
        .library(name: "Introspect-Static", type: .static, targets: ["Introspect"]),
        .library(name: "Introspect-Dynamic", type: .dynamic, targets: ["Introspect"]),
```

Libraries like [PhoneNumberKit](https://github.com/marmelroy/PhoneNumberKit/blob/master/Package.swift#L10-L12) take the same approach.

**Review by davdroman — CHANGES_REQUESTED** [review]

**Comment by davdroman** [comment]
@SplittyDev let me know if this solution looks sensible to you.

**Comment by SplittyDev** [comment]
@davdroman Generally looks good to me. I'm not sure whether the static entry is needed, though. Afaik, static linkage is the default in Swift, so adding an entry for dynamic linkage should be enough to provide both. Providing both `Introspect` and `Introspect-Static` shouldn't be necessary, unless I'm missing something.

**Comment by krzyzanowskim** [comment]
@SplittyDev `static` is the whole point of this change because automatically it happen to go dynamic and Xcode build fail

> Xcode Cloud bug that can't build and link dynamic dependency

**Comment by davdroman** [comment]
Great stuff. I'll merge this as soon as the CircleCI situation is sorted.

**Comment by SplittyDev** [comment]
> @SplittyDev `static` is the whole point of this change because automatically it happen to go dynamic and Xcode build fail
> 
> > Xcode Cloud bug that can't build and link dynamic dependency

You're absolutely right, I got this wrong at first. Great change, thanks for the PR!

**Review by davdroman — APPROVED** [review]

**Comment by davdroman** [comment]
@krzyzanowskim #179 should be fixed now. Can you push an empty commit to retrigger the CI please?

**Comment by joelpoloney** [comment]
This is no longer blocked by CircleCI FYI.

**Comment by davdroman** [comment]
Hi @krzyzanowskim. 0.2.0 is now out and includes this change. Please let us know at some point whether this fixed your Xcode Cloud issues. Thanks!

**Pull request merged**